schemas/cloudflare/dns-zone-1.yml

```diff
@@ -0,0 +1,38 @@
+---
+"$schema": /metaschema-1.json
+version: '1.0'
+type: object
+
+additionalProperties: false
+properties:
+  "$schema":
+    type: string
+    enum:
+    - /cloudflare/dns-zone-1.yml
+  labels:
+    "$ref": "/common-1.json#/definitions/labels"
+  identifier:
+    "$ref": "/common-1.json#/definitions/longIdentifier"
+  zone:
+    type: string
+  plan:
+    type: string
+    enum:
+    - free
+    - enterprise
+  type:
+    enum:
+    - full
+    - partial
+  account:
+    "$ref": "/common-1.json#/definitions/crossref"
+    "$schemaRef": "/cloudflare/account-1.yml"
+  records:
```

Comment:
How are we enforcing the 1000 record limit? Does it make sense to do this in the schema? Configuration?

Reply:
In the schema we can use a `maxItems` to enforce a maximum. The good: it fails early at schema validation, the bad: the schema error is hard to understand (bad UX). I wonder if your implementation of json schema supports custom error messages [like this](https://github.com/networknt/json-schema-validator/blob/master/doc/cust-msg.md)?

Otherwise I'm in favor of an early check in the integration as we can generally make the UX better there

Reply:
There is no need for a custom error because the jsonschema package we use in `qontract-validator` already provides a relevant error. However due to some issue in either qontract-validator's error handling that nice error is not shown and instead a more generic schema error is shown instead

I'm trying to find a way to solve this but until then I think we can still leverate the `maxItems` schema validation keyword